### ISSUE TITLE
馬場バイアス効果検証 Phase 2: 評価軸変更 + Subset 分析

### ISSUE THREAD

**Issue by daruma0411-crypto** [daruma0411-crypto/keiba-predictor#2]
## 背景

Issue #1 の Phase 1 結果は **馬券内カバー率 -0.03pt (不採用)** だったが、
これは「馬場バイアスが効かない」ではなく **実験設計の限界** が主因と判断。

### 主な問題
1. **評価軸の天井が近い** — A baseline 0.9518 (95%)。+3pt = 95→98% は構造的に厳しい
2. **粒度ミスマッチ** — bias は (date×venue×surface) 単位 → 同一日同場の12レース全てが同じ bias 値
   → race-level fixed effect として吸収されやすい

## 実装済 (commit `8067eb5`)

`scripts/simulate_v9b_phase2.py` を新規追加（Phase 1 とは別ファイル、衝突回避）。
家PC で Phase 1 にローカル修正された4点も取り込み済:
- place_code を race_id から派生
- surface を is_turf から派生
- finish rename 不要
- FEATURES_V9 を pkl 内列にフィルタ

### 追加した評価軸 (3種を並列評価)

| 指標 | 説明 | 採用候補基準 |
|---|---|---|
| Top-1 ヒット率 | 各レース mu 最小馬が1着の率 | B - A ≥ +1pt |
| 単勝 Brier | softmax(-mu/T) と1着フラグの MSE | B - A ≤ -0.001 (低いほど良) |
| オッズ加重 ROI | mu最小馬に単勝100yen, 払戻/賭金 | B - A ≥ +5pt |

### 追加した Subset 分析

| Subset | 条件 |
|---|---|
| OVERALL | 全 test set |
| BIAS-MATCHED | bias data 取得日のみ |
| BIAS-ACTIVE | matched 内で `|time_diff|>1.5` or 非フラット |
| BIAS-FLAT | matched 内で完全フラット |

仮説: bias-active subset では B - A が拡大するはず（局所効果）。

## 家PCでの実行手順

\`\`\`bash
cd C:/Users/iwash/keiba-predictor
git pull origin main
python scripts/simulate_v9b_phase2.py
\`\`\`

## 注意点

- Phase 1 スクリプト (`simulate_v9b_with_bias.py`) のローカル修正は未push 状態。
  そのまま残しておいて OK（Phase 2 とは別ファイル）。
- もし家PC側の Phase 1 修正をいずれ push したい場合は別途コミット。
- 同じ pkl・bias jsonl を流用するので追加データ準備は不要。

## 結果報告フォーマット

このissueに以下を貼ってください:

\`\`\`
[OVERALL]
A: top1=___ brier=___ roi=___
B: top1=___ brier=___ roi=___
Δ: top1=___ brier=___ roi=___

[BIAS-MATCHED]
...

[BIAS-ACTIVE]
...

[BIAS-FLAT]
...
\`\`\`

## 判定フロー

1. **OVERALL でいずれか採用候補基準クリア** → 採用、Phase 3 へ
2. **BIAS-ACTIVE でクリア & BIAS-FLAT でクリアせず** → 局所効果あり。bias_active 限定運用 or QMC統合へ
3. **どこでもクリアせず** → 馬場バイアス特徴量は捨てる。次の打ち手検討

**Comment by daruma0411-crypto** [comment]
## 実行結果（家PC 2026-04-30）

注: `features_v9b_2026.pkl` は CLAUDE.md 記載の 2026-04-11 更新前キャッシュのため FEATURES_V9 のうち 4 列 (`avg_jyuni_1c`, `avg_jyuni_2c`, `has_same_dist`, `has_long_stretch`) が欠落。**30/34 特徴量**で A/B 共通実行（公平比較）。

### 共通指標
- matched: 274021 / 528721 (51.8%)
- train: 191,027 / test: 42,432（芝のみ）

### 結果

```
[OVERALL]
        model  n_races  top5_top3  top1_hit  brier     roi
 A (baseline)     3109     0.9518    0.2042 0.0694 -0.1388
B (with bias)     3109     0.9514    0.2084 0.0681 -0.2155
  Delta: top5_top3 -0.0003 | top1 +0.0042 | brier -0.0013 | roi -0.0767

[BIAS-MATCHED]
        model  n_races  top5_top3  top1_hit  brier     roi
 A (baseline)     2845     0.9508    0.2074 0.0696 -0.1053
B (with bias)     2845     0.9494    0.2063 0.0684 -0.2135
  Delta: top5_top3 -0.0014 | top1 -0.0011 | brier -0.0012 | roi -0.1082

[BIAS-ACTIVE]   (n=2159)
        model  n_races  top5_top3  top1_hit  brier     roi
 A (baseline)     2159     0.9500    0.2061 0.0698 -0.1126
B (with bias)     2159     0.9518    0.2006 0.0691 -0.2504
  Delta: top5_top3 +0.0019 | top1 -0.0056 | brier -0.0007 | roi -0.1378

[BIAS-FLAT]   (n=686)
        model  n_races  top5_top3  top1_hit  brier     roi
 A (baseline)      686     0.9534    0.2114 0.0689 -0.0827
B (with bias)      686     0.9417    0.2245 0.0664 -0.0974
  Delta: top5_top3 -0.0117 | top1 +0.0131 | brier -0.0025 | roi -0.0147
```

### 採用基準クリア状況

| Subset | top1 ≥+0.01 | brier ≤-0.001 | roi ≥+0.05 |
|---|:---:|:---:|:---:|
| OVERALL | ✗ (+0.0042) | ✓ (-0.0013) | ✗ (-0.0767) |
| BIAS-MATCHED | ✗ (-0.0011) | ✓ (-0.0012) | ✗ (-0.1082) |
| BIAS-ACTIVE | ✗ (-0.0056) | ✗ (-0.0007) | ✗ (-0.1378) |
| BIAS-FLAT | ✓ (+0.0131) | ✓ (-0.0025) | ✗ (-0.0147) |

### 観察と判定

1. **仮説と真逆の結果**: BIAS-ACTIVE では効果なし／悪化。BIAS-FLAT で逆に B が改善。
   仮説（active subset で効果が拡大）は否定された。
2. **brier は全 subset で改善**: 確率校正は良くなっているが Top1・ROI には伝播していない。NN がノイズを吸収した結果の校正向上の可能性。
3. **ROI は全 subset で B が悪化**: 実用面では明確にネガティブ。
4. **BIAS-FLAT で B が改善する解釈**: 「フラット」というシグナル自体が「普通の馬場で実力通り」を意味するため、NN が安定領域を識別する材料として使えている可能性。逆に BIAS-ACTIVE では bias 値だけで馬個別の適性差を表現できないため噪音化。

### 判定フローへのマッピング

- OVERALL で **brier 1 指標のみクリア**（top1・roi は未達）
- BIAS-ACTIVE は全滅。BIAS-FLAT で 2 指標クリア（仮説と逆）

→ 判定フロー上は「OVERALL で 1 指標クリア」だが、ROI 全面悪化＋仮説否定により**実用採用は不可**。
**3. どこでもクリアせず（実質）→ 馬場バイアス特徴量は捨てる、または別アプローチ検討**。

### 次の打ち手候補

1. **粒度を上げる**: bias を (date×venue×surface) ではなく **(date×venue×surface×馬個別属性)** に拡張（枠順×frame_bias の交互作用、脚質×fb_bias など）。これなら同一レース内で馬ごとに値が変わる。
2. **NN 統合をやめて Layer 2 QMC のコース別補正に組み込む**: bias 値で枠順優劣・前後傾向の事前分布を変える。NN 学習時には使わない。
3. **キャッシュ再構築後の再検証**: 4 列欠落の影響度確認のため再構築後に Phase 1+2 を再実行（影響軽微の可能性が高いが念のため）。
4. **馬場バイアス特徴量を棄却**: 残りリソースを次の効果的な特徴量探索に振る。

推奨: **2 → 1 の順**。Layer 2 統合は Phase 1+2 で確認できた「NN 入力としては機能しない」結論と整合し、bias 情報が本来生きる場所（コース物理特性 + 当日条件）への配置として理にかなっている。
